`std.posix.sendfile()` fails to compile on 32-bit FreeBSD

For `arm-freebsd.14.0-eabihf`:

```
/home/alexrp/Source/ziglang/zig-bootstrap/out/host/lib/zig/std/posix.zig:6470:36: error: @bitCast size mismatch: destination type 'usize' has 32 bits but source type 'i64' has 64 bits
                const amt: usize = @bitCast(sbytes);
                                   ^~~~~~~~~~~~~~~~
```

--- a/test/tests.zig
+++ b/test/tests.zig
@@ -116,8 +116,6 @@ const test_targets = blk: {
                 .abi = .eabihf,
             },
             .link_libc = true,
-            // https://github.com/ziglang/zig/issues/23949
-            .skip_modules = &.{"std"},
         },
 
         .{

